ESQL: SingleValueMatchQueryTests failing in release tests

### CI Link

https://gradle-enterprise.elastic.co/s/5lbobv2xwqkpg/tests/task/:x-pack:plugin:esql:test/details/org.elasticsearch.xpack.esql.querydsl.query.SingleValueMatchQueryTests/testQuery%20%7BStandardSetup%5BfieldType%3Dkeyword%2C%20multivaluedField%3Dtrue%2C%20docValuesMode%3DDOC_VALUES_ONLY_HIGH_CARDINALITY%2C%20empty%3Dfalse%2C%20count%3D100%5D%7D?top-execution=1

### Repro line

REPRODUCE WITH: ./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.querydsl.query.SingleValueMatchQueryTests.testQuery {StandardSetup[fieldType=keyword, multivaluedField=true, docValuesMode=DOC_VALUES_ONLY_HIGH_CARDINALITY, empty=false, count=100]}" -Dtests.seed=AB2183C68CF332BB -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=en-VC -Dtests.timezone=Pacific/Galapagos -Druntime.java=25

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

```
org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: Could not convert [doc_values] to boolean	
at __randomizedtesting.SeedInfo.seed([AB2183C68CF332BB:2050F7B57D79086E]:0)	
at org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:672)	
at org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:619)	
at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:615)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.merge(MapperServiceTestCase.java:507)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.merge(MapperServiceTestCase.java:489)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.withMapping(MapperServiceTestCase.java:257)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:237)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:224)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:185)	
at org.elasticsearch.xpack.esql.querydsl.query.SingleValueMatchQueryTests.testQuery(SingleValueMatchQueryTests.java:93)	
at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)	
at java.lang.reflect.Method.invoke(Method.java:565)	
at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1763)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)	
at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)	
at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)	
at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)	
at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)	
at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)	
at org.elasticsearch.entitlement.bootstrap.TestEntitlementsRule$1.evaluate(TestEntitlementsRule.java:77)	
at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)	
at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)	
at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)	
at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)	
at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)	
at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)	
at java.lang.Thread.run(Thread.java:1474)	
Caused by: java.lang.IllegalArgumentException: Could not convert [doc_values] to boolean	
at org.elasticsearch.common.xcontent.support.XContentMapValues.nodeBooleanValue(XContentMapValues.java:523)	
at org.elasticsearch.index.mapper.FieldMapper$DocValuesParameter.parse(FieldMapper.java:1449)	
at org.elasticsearch.index.mapper.FieldMapper$Builder.parse(FieldMapper.java:1714)	
at org.elasticsearch.index.mapper.FieldMapper$TypeParser.parse(FieldMapper.java:1856)	
at org.elasticsearch.index.mapper.FieldMapper$TypeParser.parse(FieldMapper.java:1806)	
at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:413)	
at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:328)	
at org.elasticsearch.index.mapper.RootObjectMapper.parse(RootObjectMapper.java:476)	
at org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:125)	
at org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:670)	
at org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:619)	
at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:615)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.merge(MapperServiceTestCase.java:507)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.merge(MapperServiceTestCase.java:489)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.withMapping(MapperServiceTestCase.java:257)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:237)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:224)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:185)	
at org.elasticsearch.xpack.esql.querydsl.query.SingleValueMatchQueryTests.testQuery(SingleValueMatchQueryTests.java:93)	
at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)	
at java.lang.reflect.Method.invoke(Method.java:565)	
at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1763)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)	
at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)	
at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)	
at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)	
at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)	
at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)	
at org.elasticsearch.entitlement.bootstrap.TestEntitlementsRule$1.evaluate(TestEntitlementsRule.java:77)	
at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)	
at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)	
at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)	
at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)	
at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)	
at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)	
at java.lang.Thread.run(Thread.java:1474)	
Caused by: java.lang.IllegalArgumentException: Failed to parse value [{cardinality=high}] as only [true] or [false] are allowed.	
at org.elasticsearch.core.Booleans.parseBoolean(Booleans.java:63)	
at org.elasticsearch.common.xcontent.support.XContentMapValues.nodeBooleanValue(XContentMapValues.java:531)	
at org.elasticsearch.common.xcontent.support.XContentMapValues.nodeBooleanValue(XContentMapValues.java:521)	
at org.elasticsearch.index.mapper.FieldMapper$DocValuesParameter.parse(FieldMapper.java:1449)	
at org.elasticsearch.index.mapper.FieldMapper$Builder.parse(FieldMapper.java:1714)	
at org.elasticsearch.index.mapper.FieldMapper$TypeParser.parse(FieldMapper.java:1856)	
at org.elasticsearch.index.mapper.FieldMapper$TypeParser.parse(FieldMapper.java:1806)	
at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:413)	
at org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:328)	
at org.elasticsearch.index.mapper.RootObjectMapper.parse(RootObjectMapper.java:476)	
at org.elasticsearch.index.mapper.MappingParser.parse(MappingParser.java:125)	
at org.elasticsearch.index.mapper.MapperService.parseMapping(MapperService.java:670)	
at org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:619)	
at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:615)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.merge(MapperServiceTestCase.java:507)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.merge(MapperServiceTestCase.java:489)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.withMapping(MapperServiceTestCase.java:257)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:237)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:224)	
at org.elasticsearch.index.mapper.MapperServiceTestCase.createMapperService(MapperServiceTestCase.java:185)	
at org.elasticsearch.xpack.esql.querydsl.query.SingleValueMatchQueryTests.testQuery(SingleValueMatchQueryTests.java:93)	
at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)	
at java.lang.reflect.Method.invoke(Method.java:565)	
at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1763)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)	
at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)	
at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)	
at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)	
at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)	
at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)	
at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)	
at org.elasticsearch.entitlement.bootstrap.TestEntitlementsRule$1.evaluate(TestEntitlementsRule.java:77)	
at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)	
at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)	
at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)	
at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)	
at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)	
at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)	
at org.junit.rules.RunRules.evaluate(RunRules.java:20)	
at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)	
at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)	
at java.lang.Thread.run(Thread.java:1474)
```

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java
@@ -19,6 +19,8 @@ import java.util.List;
 import java.util.Locale;
 import java.util.Set;
 
+import static org.elasticsearch.index.mapper.FieldMapper.DocValuesParameter.EXTENDED_DOC_VALUES_PARAMS_FF;
+
 /**
  * A {@link Set} of "capabilities" supported by the {@link RestEsqlQueryAction}
  * and {@link RestEsqlAsyncQueryAction} APIs. These are exposed over the
@@ -809,6 +811,11 @@ public class EsqlCapabilities {
          * _source field mapping directives: https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-source-field.html
          */
         SOURCE_FIELD_MAPPING,
+
+        /**
+         * Support for extended doc values params.
+         */
+        EXTENDED_DOC_VALUES_PARAMS(EXTENDED_DOC_VALUES_PARAMS_FF.isEnabled()),
 
         /**
          * Allow filter per individual aggregation.

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueMatchQueryTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueMatchQueryTests.java
@@ -25,7 +25,6 @@ import org.apache.lucene.store.Directory;
 import org.apache.lucene.tests.index.RandomIndexWriter;
 import org.apache.lucene.util.BytesRef;
 import org.apache.lucene.util.NumericUtils;
-import org.elasticsearch.Build;
 import org.elasticsearch.compute.operator.DriverContext;
 import org.elasticsearch.compute.operator.Warnings;
 import org.elasticsearch.compute.querydsl.query.SingleValueMatchQuery;
@@ -42,6 +41,7 @@ import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
+import static org.elasticsearch.index.mapper.FieldMapper.DocValuesParameter.EXTENDED_DOC_VALUES_PARAMS_FF;
 import static org.elasticsearch.index.mapper.MultiValuedBinaryDocValuesField.SeparateCount.COUNT_FIELD_SUFFIX;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.instanceOf;
@@ -66,7 +66,7 @@ public class SingleValueMatchQueryTests extends MapperServiceTestCase {
                     for (DocValuesMode docValuesMode : new DocValuesMode[] { DocValuesMode.DEFAULT, DocValuesMode.DOC_VALUES_ONLY }) {
                         params.add(new Object[] { new StandardSetup(fieldType, multivaluedField, docValuesMode, allowEmpty, 100) });
                     }
-                    if (fieldType.equals("keyword") && Build.current().isSnapshot()) {
+                    if (fieldType.equals("keyword") && EXTENDED_DOC_VALUES_PARAMS_FF.isEnabled()) {
                         params.add(
                             new Object[] {
                                 new StandardSetup(

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueMatchQueryTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/querydsl/query/SingleValueMatchQueryTests.java
@@ -25,6 +25,7 @@ import org.apache.lucene.store.Directory;
 import org.apache.lucene.tests.index.RandomIndexWriter;
 import org.apache.lucene.util.BytesRef;
 import org.apache.lucene.util.NumericUtils;
+import org.elasticsearch.Build;
 import org.elasticsearch.compute.operator.DriverContext;
 import org.elasticsearch.compute.operator.Warnings;
 import org.elasticsearch.compute.querydsl.query.SingleValueMatchQuery;
@@ -65,7 +66,7 @@ public class SingleValueMatchQueryTests extends MapperServiceTestCase {
                     for (DocValuesMode docValuesMode : new DocValuesMode[] { DocValuesMode.DEFAULT, DocValuesMode.DOC_VALUES_ONLY }) {
                         params.add(new Object[] { new StandardSetup(fieldType, multivaluedField, docValuesMode, allowEmpty, 100) });
                     }
-                    if (fieldType.equals("keyword")) {
+                    if (fieldType.equals("keyword") && Build.current().isSnapshot()) {
                         params.add(
                             new Object[] {
                                 new StandardSetup(

--- a/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/250_high_cardinality_keyword.yml
+++ b/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/250_high_cardinality_keyword.yml
@@ -1,7 +1,12 @@
 ---
 setup:
   - requires:
-      cluster_features: ["mapper.keyword.high_cardinality_length_function_fuse_to_load"]
+      test_runner_features: [ capabilities ]
+      capabilities:
+        - method: POST
+          path: /_query
+          parameters: [ ]
+          capabilities: [ extended_doc_values_params ]
       reason: "testing binary doc values search"
 
   - do:

